tunnel-server/src/proxy.ts

```diff
@@ -52,6 +58,24 @@ export function proxyHandlers({
         return;
       }
 
+      if(env.access === 'private'){
+        const authorization = req.headers['authorization'];
+        if (!authorization) {
+          return unauthorized(res)
+        }
+
+        const basicAuth = Buffer.from(authorization?.split('Basic ')[1] || '', 'base64').toString('ascii');
+        const [username, password] = basicAuth.split(':');
+        if (username !== 'x-preevy-key'){
+          return unauthorized(res)
+        }
+        const signature = Buffer.from(password, 'base64');
```

Comment:
If the data is already wrapped in base64, is there any text format that can be stored binary data more efficiently and still passed in clear text and in url?

Reply:
for arbitrary binary data base64 is the most space efficient